src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -1001,8 +1003,9 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VariantMetadata,
 
   def deleteVA(path: List[String]): (Type, Deleter) = vaSignature.delete(path)
 
-  def downsampleVariants(keep: Long): VariantSampleMatrix[T] = {
-    sampleVariants(keep.toDouble / countVariants())
+  def downsampleVariants(keep: Double): VariantSampleMatrix[T] = {
```

Comment:
You changed "keep" to "fraction" in python, you should change it here as well (both for consistency and because it's confusing that the require message's reference to "keep" is unclear to a python user).